msal/src/Microsoft.Identity.Client/ClientApplicationBase.cs

```diff
@@ -52,12 +53,26 @@ public abstract partial class ClientApplicationBase
         protected const string DefaultAuthority = "https://login.microsoftonline.com/common/";
 
         /// <summary>
-        /// 
+        /// Constructor of the base application
         /// </summary>
-        /// <param name="clientId"></param>
-        /// <param name="authority"></param>
-        /// <param name="redirectUri"></param>
-        /// <param name="validateAuthority"></param>
+        /// <param name="clientId">Client ID (also named Application ID) of the application as registered in the 
+        /// application registration portal (https://aka.ms/msal-net-register-app)</param>
+        /// <param name="authority">Authority of the Security service token (STS) from which MSAL.NET will acquire the tokens.
+        /// 
+        /// Usual authorities are:
+        /// <list type="bullet">
+        /// <item><c>https://login.microsoftonline.com/tenant/</c>, where <c>tenant</c> is the tenant ID of the Azure AD tenant
+        /// or a domain associated with this Azure AD tenant, in order to sign-in user of a specific organization only</item>
+        /// <item><c>https://login.microsoftonline.com/common/</c> to signing users with any work and school accounts or Microsoft personal account</item>
+        /// <item><c>https://login.microsoftonline.com/organizations/</c> to signing users with any work and school accounts</item>
+        /// <item><c>https://login.microsoftonline.com/consumers/</c> to signing users with only personal Microsoft account (live)</item>
+        /// </list>
+        /// Note that this setting needs to be consistent with what is declared in the application registration portal
+        /// </param>
+        /// <param name="redirectUri">URL where the STS will call back the application with the security token</param>
```

Comment:
>URL where the STS [](start = 38, length = 17)

This is a rather tricky area and expecially with both System  Browser and embedded webview challenges (for native apps). Should we have an aka.ms link to explain more about redirectUri's?

Reply:
Added an aka.ms link (as well as in all the Application classes)